core/src/test/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassMonitorSpec.scala

```diff
@@ -0,0 +1,381 @@
+    val terraWorkspaceNextflowRole = "fakeTerraWorkspaceNextflowRole"
+    val terraBucketReaderRole = "fakeTerraBucketReaderRole"
+    val terraBucketWriterRole = "fakeTerraBucketWriterRole"
+
+    val fastPassConfig =
+      FastPassConfig.apply(testConf).copy(enabled = fastPassEnabled).copy(grantPeriod = fastPassGrantPeriod)
+    val fastPassServiceConstructor = FastPassService.constructor(
+      fastPassConfig,
+      googleIamDAO,
+      googleStorageDAO,
+      samDAO,
+      terraBillingProjectOwnerRole,
+      terraWorkspaceCanComputeRole,
+      terraWorkspaceNextflowRole,
+      terraBucketReaderRole,
+      terraBucketWriterRole
+    ) _
+
+    val workspaceServiceConstructor = WorkspaceService.constructor(
+      slickDataSource,
+      new HttpMethodRepoDAO(
+        MethodRepoConfig[Agora.type](mockServer.mockServerBaseUrl, ""),
+        MethodRepoConfig[Dockstore.type](mockServer.mockServerBaseUrl, ""),
+        workbenchMetricBaseName = workbenchMetricBaseName
+      ),
+      new HttpExecutionServiceDAO(mockServer.mockServerBaseUrl, workbenchMetricBaseName = workbenchMetricBaseName),
+      executionServiceCluster,
+      execServiceBatchSize,
+      workspaceManagerDAO,
+      methodConfigResolver,
+      gcsDAO,
+      samDAO,
+      notificationDAO,
+      userServiceConstructor,
+      genomicsServiceConstructor,
+      maxActiveWorkflowsTotal,
+      maxActiveWorkflowsPerUser,
+      workbenchMetricBaseName,
+      submissionCostService,
+      workspaceServiceConfig,
+      requesterPaysSetupService,
+      entityManager,
+      resourceBufferService,
+      resourceBufferSaEmail,
+      servicePerimeterService,
+      googleIamDAO,
+      terraBillingProjectOwnerRole,
+      terraWorkspaceCanComputeRole,
+      terraWorkspaceNextflowRole,
+      terraBucketReaderRole,
+      terraBucketWriterRole,
+      rawlsWorkspaceAclManager,
+      multiCloudWorkspaceAclManager,
+      fastPassServiceConstructor
+    ) _
+
+    def cleanupSupervisor =
+      submissionSupervisor ! PoisonPill
+  }
+
+  def withTestDataServicesFastPassGrantPeriodZero[T](testCode: TestApiService => T) =
+    withDefaultTestDatabase { dataSource: SlickDataSource =>
+      withServices(dataSource, testData.userOwner, fastPassGrantPeriod = java.time.Duration.ofSeconds(0))(testCode)
+    }
+
+  def withServices[T](dataSource: SlickDataSource,
+                      user: RawlsUser,
+                      fastPassEnabled: Boolean = true,
+                      fastPassGrantPeriod: java.time.Duration = java.time.Duration.ofHours(2)
+  )(
+    testCode: (TestApiService) => T
+  ) = {
+    val apiService = new TestApiService(dataSource, user, fastPassEnabled, fastPassGrantPeriod)
+    try
+      testCode(apiService)
+    finally
+      apiService.cleanupSupervisor
+  }
+
+  "FastPassMonitor" should "remove expired fastpass grants" in withTestDataServicesFastPassGrantPeriodZero { services =>
```

Comment:
I'd like to see a test that checks that a single troublesome grant doesn't block up the entire monitor

Reply:
I've added a test in the latest commit.